raster/src/main/scala/geotrellis/raster/io/geotiff/reader/GeoTiffStreamReader.scala

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (c) 2014 Azavea.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster.io.geotiff.reader
+
+import geotrellis.raster._
+import geotrellis.raster.io.geotiff._
+import geotrellis.raster.io.geotiff.compression._
+import geotrellis.raster.io.geotiff.util._
+import geotrellis.raster.io.geotiff.tags._
+import geotrellis.vector.Extent
+import geotrellis.proj4.CRS
+import geotrellis.util.Filesystem
+
+import monocle.syntax.apply._
+
+import scala.io._
+import scala.collection.mutable
+import java.nio.{ByteBuffer, ByteOrder}
+import spire.syntax.cfor._
+
+
+object GeoTiffStreamReader {
```

Comment:
We shouldn't have a mostly duplicated object for this.

We should be reading tiff tags from a byte buffer, and that byte buffer should be backed by a file map when reading the tifftags from a path (and not as part of the GeoTiff reading process).

Streaming reads are not useful if we're reading in the whole raster. It will be useful when we are reading a window out, but we're not there yet.